### PR TITLE
[dnssd] add tests for Service Instance names with spaces

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit adds tests to make sure Service Instance names with spaces are handled properly by SRP and DNS-SD.

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+DOMAIN = 'default.service.arpa.'
+SERVICE = '_testsrv._udp'
+SERVICE_FULL_NAME = f'{SERVICE}.{DOMAIN}'
+INSTANCE_NAME = 'O T B R'
+HOST_NAME = 'host1'
+HOST_FULL_NAME = f'{HOST_NAME}.{DOMAIN}'
+
+
+class Test(thread_cert.TestCase):
```
Comment (superwhd) [inline tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py:60]: Use a more specific test class name?

File: tests/scripts/thread-cert/node.py [hunk 2]
```diff
@@ -292,17 +293,48 @@ def dns_dig(self, server: str, name: str, qtype: str):
...
+    @staticmethod
+    def __unescape_dns_instance_name(name: str) -> str:
+        new_name = []
+        i = 0
+        while i < len(name):
+            c = name[i]
+
+            if c == '\\':
+                assert i + 1 < len(name), name
+                if name[i + 1] in string.digits:
```
[the remark this answers is not in the document]
Comment (simonlingoogle) [reply tests/scripts/thread-cert/node.py:323]: Thanks. `.isdigit()` should be much more efficient.

File: tests/scripts/thread-cert/node.py [hunk 3]
```diff
@@ -292,17 +293,48 @@ def dns_dig(self, server: str, name: str, qtype: str):
...
+                    assert i + 3 < len(name) and name[i + 2] in string.digits and name[i + 3] in string.digits, name
+                    b = (ord(name[i + 1]) - 48) * 100 + (ord(name[i + 2]) - 48) * 10 + (ord(name[i + 3]) - 48)
```
[the remark this answers is not in the document]
Comment (simonlingoogle) [reply tests/scripts/thread-cert/node.py:325]: I wonder why I wrote it.

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**